src/main/java/com/faforever/client/chat/ChatUserControl.java

```diff
@@ -76,17 +115,115 @@ public PlayerInfoBean getPlayerInfoBean() {
+          String updatedLadderRating = i18n.get("playerRatingFormat.ladder", RatingUtil.getLadderRating(playerInfoBean));
+          userRatingTooltip.setText(String.format("%s \n %s", oldGlobalRating, updatedLadderRating));
+        });
+
+        Tooltip.install(clanLabel, userRatingTooltip);
+        Tooltip.install(usernameLabel, userRatingTooltip);
+      }
+    });
+
+  }
+
+  private void configureColor() {
+    ChatPrefs chatPrefs = preferencesService.getPreferences().getChatPrefs();
+    ObjectProperty<Color> colorProperty = null;
+    Color color = null;
+
+    //Always regardless of pretty Colors enabled
+    if (playerInfoBean.getModeratorInChannels().size() > 0) {
+      colorProperty = chatPrefs.modsChatColorProperty();
+    } else if (playerInfoBean.isFriend()) {
+      colorProperty = chatPrefs.friendsChatColorProperty();
+    } else if (playerInfoBean.isFoe()) {
+      colorProperty = chatPrefs.foesChatColorProperty();
+    } else if (playerInfoBean.getUsername().equals(userService.getUsername())) {
+      colorProperty = chatPrefs.selfChatColorProperty();
+    }
+
+    //Only if pretty colors is disabled
+    if (!chatPrefs.getPrettyColors() && colorProperty == null) {
+      if (playerInfoBean.isChatOnly()) {
+        colorProperty = chatPrefs.ircChatColorProperty();
+      } else {
+        colorProperty = chatPrefs.othersChatColorProperty();
+      }
+
+      //Self color
+    } else if (chatPrefs.getPrettyColors() && colorProperty == null) {
+      color = ColorGeneratorUtil.generatePrettyHexColor();
+      ChatUser chatUser = chatService.getChatUser(playerInfoBean.getUsername());
+
+      try {
+        if (chatUser.getColor() == null) {
+          chatUser.setColor(color);
+        } else {
+          color = chatUser.getColor();
+        }
+      } catch (NullPointerException e) {
+        logger.warn("Could not generate color for {}", playerInfoBean.getUsername());
+      }
+    }
+
+    if (colorProperty != null) {
+      //FIXME if there is a listener already will this break?
+      color = colorProperty.get();
+      addColorListenerToLabels(colorProperty);
+    }
+
+    usernameLabel.setTextFill(color);
+    clanLabel.setTextFill(color);
+  }
+
+  private void addColorListenerToLabels(ObjectProperty<Color> colorProperty) {
+    colorProperty.addListener((observable, oldValue, newValue) -> {
+      usernameLabel.setTextFill(newValue);
+      clanLabel.setTextFill(newValue);
+    });
+  }
+
+  private void addPrettyColorListener() {
+    ChatPrefs chatPrefs = preferencesService.getPreferences().getChatPrefs();
+    chatPrefs.prettyColorsProperty().addListener((observable, oldValue, newValue) -> {
+      configureColor();
+    });
+  }
+
   private void configureCountryImageView() {
-    playerInfoBean.countryProperty().addListener((observable, oldValue, newValue) -> {
+    //FIXME is this really necessary, when will someone's country change in a single session?
```

Comment:
If I log in and then travel to germany, you'd still see me as "swiss" :-) Or if I use a proxy, or.. I'd just do it.